src/app/room/room.cpp

```diff
@@ -14,20 +15,29 @@ Room::Room(const unsigned room_id, const std::shared_ptr<Warehouse>& warehouse)
 {
 }
 
-bool Room::isVisitAssigned(const std::shared_ptr<Visit>& visit)
+bool Room::isVisitFound(const std::shared_ptr<Visit>& visit)
 {
-    return std::find(visits_.begin(), visits_.end(), visit) == visits_.end() ? false : true;
+    return std::find(visits_.begin(), visits_.end(), visit) != visits_.end();
 }
 
 void Room::addVisit(const std::shared_ptr<Visit>& visit)
 {
-    if (!isVisitAssigned(visit))
+    if (!visit)
+    {
+        throw std::invalid_argument(std::string("Argument points to nullptr in ") + __func__);
+    }
+    if (!isVisitFound(visit))
     {
         visits_.emplace_back(visit);
         visit->setRoom(shared_from_this());
     }
 }
 
+void Room::removeVisit(const std::shared_ptr<Visit>& visit)
+{
+    visits_.erase(std::remove(visits_.begin(), visits_.end(), visit), visits_.end());
```

Comment:
In one of the unit tests I encountered memory issues when removed like that.
I am not saying to correct this at this point, but as potential point to check in case of memory problems.